test/e2e/kubectl.go

```diff
@@ -0,0 +1,215 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	nautilusImage       = "kubernetes/update-demo:nautilus"
+	kittenImage         = "kubernetes/update-demo:kitten"
+	updateDemoSelector  = "name=update-demo"
+	updateDemoContainer = "update-demo"
+)
+
+var _ = Describe("kubectl", func() {
+
+	runUpdateExample()
+})
+
+func runUpdateExample() {
+	updateDemoRoot := filepath.Join(root, "examples/update-demo")
+
+	It(fmt.Sprint("should create and stop a replication controller"), func() {
+		nautilusPath := filepath.Join(updateDemoRoot, "nautilus-rc.yaml")
+
+		defer cleanup(nautilusPath, updateDemoSelector)
+		By("creating a replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(updateDemoSelector, updateDemoContainer, nautilusImage, 2, 30*time.Second)
+	})
+
+	It(fmt.Sprint("should scale a replication controller"), func() {
+		nautilusPath := filepath.Join(updateDemoRoot, "nautilus-rc.yaml")
+
+		defer cleanup(nautilusPath, updateDemoSelector)
+		By("creating a replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(updateDemoSelector, updateDemoContainer, nautilusImage, 2, 30*time.Second)
+		By("scaling down the replication controller")
+		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
+		validateController(updateDemoSelector, updateDemoContainer, nautilusImage, 1, 30*time.Second)
+		By("scaling up the replication controller")
+		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
+		validateController(updateDemoSelector, updateDemoContainer, nautilusImage, 2, 30*time.Second)
+	})
+
+	It(fmt.Sprint("should do a rolling update of a replication controller"), func() {
+		nautilusPath := filepath.Join(updateDemoRoot, "nautilus-rc.yaml")
+		kittenPath := filepath.Join(updateDemoRoot, "kitten-rc.yaml")
+
+		// Cleanup all resources incase we fail somewhere in the middle
+		defer cleanup(updateDemoRoot, updateDemoSelector)
+
+		By("creating the initial replication controller")
+		runKubectl("create", "-f", nautilusPath)
+		validateController(updateDemoSelector, updateDemoContainer, nautilusImage, 2, 30*time.Second)
+		By("rollingupdate to new replication controller")
+		runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
+		validateController(updateDemoSelector, updateDemoContainer, kittenImage, 2, 30*time.Second)
+	})
+
+}
+
+func cleanup(filePath, selector string) {
```

Comment:
Every call to cleanup passes updateDemoSelector as the 2nd argument. Do we really need to parameterize it as an argument?